mlflow/pyfunc/__init__.py

```diff
@@ -716,6 +719,64 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl)
 
 
+def _download_model_conda_env(model_uri):
+    model_meta_path = _download_artifact_from_uri(os.path.join(model_uri, MLMODEL_FILE_NAME))
+    model_meta = Model.load(model_meta_path)
+    conda_yml_file_name = model_meta.flavors[FLAVOR_NAME][ENV]
+    return _download_artifact_from_uri(os.path.join(model_uri, conda_yml_file_name))
+
+
+def get_model_dependencies(model_uri, format="pip"):
```

Comment:
@harupy @WeichenXu123 Note that this doesn't handle the case where `conda.yaml`'s `pip` section includes `-r requirements.txt` references. It also doesn't handle the case where a user manually constructs a `requirements.txt` file in the MLflow Model that references other `requirements.txt` files.

I don't think we need to handle either of these cases. In MLflow 2.0, I think we should implement some additional restrictions on the accepted pip requirements / conda + pip requirements to disallowed `-r`. In MLflow 2.0, we should also expand the pyfunc flavor's `env` field to include multiple environment specs: pip `requirements.txt` and Conda `conda.yaml`; it's currently a bit strange that `env` doesn't include `requirements.txt`.